File: Leetcode/LargestNumber.cpp
class Solution {
public:
    static bool cmp(string &s1, string &s2){
	return s1 + s2 > s2 + s1;
}

string largestNumber(vector<int>& nums) {
	vector<string> s;

	for (int i = 0; i < nums.size(); ++i){
		stringstream ss;
		ss << nums[i];
		s.push_back(string(ss.str()));
	}
	sort(s.begin(), s.end(), cmp);
	string ans = "";
	for (int i = 0; i < s.size(); ++i) ans += s[i];
	string res = "";
	for(int i = 0 ; i < ans.size(); ++i){
	    if(ans[i] != '0'){
	        res.insert(0, ans, i, ans.size() - i);
	        break;
	    } 
	}
	return (res.empty() ? "0" : ans);
    }
};

File: Leetcode/LargestNumber.rs
// https://leetcode.com/problems/largest-number

struct Solution {}

impl Solution {
    pub fn largest_number(mut nums: Vec<i32>) -> String {
        nums.sort_by(|&a, &b| {
            let ab = format!("{a}{b}");
            let ba = format!("{b}{a}");

            ba.cmp(&ab)
        });

        let s = nums.into_iter().fold(String::new(), |mut s, n| {
            s.push_str(&format!("{n}"));
            s
        });
        let s = s.trim_start_matches('0');

        if s.is_empty() {
            String::from("0")
        } else {
            s.to_owned()
        }
    }
}
